File: backend/app/app/schemas/payment.py
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import PaymentMethod, PaymentStatus


class DummyPaymentApproveRequest(BaseModel):
    order_id: int
    idempotency_key: str | None = None


class PaymentRead(BaseModel):
    id: int
    order_id: int
    status: PaymentStatus
    method: PaymentMethod
    approved_amount: int
    dummy_approval_code: str | None = None
    approved_at: datetime | None = None

File: shared/frontend-client/src/types/payment.ts
export type PaymentStatus = "pending" | "approved" | "failed" | "refunded";

export type PaymentMethod = "dummy_card" | "point";

export interface DummyPaymentApproveRequest {
  order_id: number;
  idempotency_key?: string | null;
  simulate_failure?: boolean;
}

export interface PaymentRead {
  id: number;
  order_id: number;
  status: PaymentStatus;
  method: PaymentMethod;
  approved_amount: number;
  dummy_approval_code: string | null;
  approved_at: string | null;
}
